lib/Importer/ONNXModelLoader.cpp

```diff
@@ -36,14 +36,29 @@ using namespace glow;
 using llvm::cast;
 
 namespace {
+
+/// Get the shape of a TensorShapeProto given by \p shapeProto and return the
+/// dimensions in the vector \p dim passed by reference.
+Error getProtoShape(const ONNX_NAMESPACE::TensorShapeProto &shapeProto,
+                    std::vector<size_t> &dim) {
+  for (auto d : shapeProto.dim()) {
+    // If the proto has a symbolic dimension "dim_param" we will force it to 1.
```

Comment:
This seems kind of like a hack. I'm wondering if we should make this opt-in, maybe through a cl opt, or cmake flag to configure Glow with if cl opt doesn't work for you. WDYT?

Reply:
It looks like a hack but I wouldn`t treat like one. A lot of the models obtained through conversion from TensorFlow to ONNX exhibit placeholders with the batch size defined as a "dim_param" with some funny value (like "None" or "unk_symbol" etc). For these kind of models Glow currently retrieves an error because Glow does not support the "dim_param" field which is intended for symbolic and implicitly dynamic sizes which Glow does not support (and most probably never will).

Reply:
Again you can take a look at the example #3791.
If it feels hackish for you then you what should be the elegant solution?

Reply:
So I pushed another commit in which I added a command line option called `-onnx-define-symbol` which allows the user to define symbols in order to resolve undefined symbols from the model. In order to make it less hackish (as you suggested) I made it mandatory, in the sense that if there`s an unresolved symbol in the model I don`t force it by default to a value of 1 but instead I expect the user to define it (otherwise I throw an error).

The syntax of this command line options is: `-onnx-define-symbol=name,value`. For example in my case I use it as: `-onnx-define-symbol=None,1`. Multiple symbols can be defined by using this option multiple times.

What do you say?